Bug Fix Replaced UTF-8 NO-BREAK SPAC character with regular space

jongeller opened the pull request:
This bug fix updates the renderer code and modifies the MAML renderer to use a new sanitization class RenderCleaner to replace the special whitespace characters with a regular space.

It also abstracts the line break sanitization in Markdown generation to the RenderCleaner class, as this was exclusively used by the MarkDown render for V2 cmdlets markdown. It was abstracted because this may have more widely applicable usage as we add features from the road map, such as AboutTopic generation and rendering.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/platyps/116)
<!-- Reviewable:end -->

vors commented on src/Markdown.MAML/Renderer/RenderCleaner.cs:
```diff
@@ -0,0 +1,24 @@
+{
+    static class RenderCleaner
+    {
+        public static string NormalizeCharacters(string text)
```
Please, rename to NormalizeWhitespaces

jongeller commented:
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/Markdown.MAML/Renderer/RenderCleaner.cs, line 13 \[r1\]](https://reviewable.io:443/reviews/powershell/platyps/116#-KJGVKkmg8RfcFhGeL_N-r1-13:-KJHkW9lYwAT-iahFB82:-2079945895) ([raw file](https://github.com/powershell/platyps/blob/a7ec110052b8b6b921d5fc8392fa007b3ffd0a09/src/Markdown.MAML/Renderer/RenderCleaner.cs#L13)):*
<details><summary><i>Previously, vors (Sergei Vorobev) wrote…</i></summary>
> Please, rename to NormalizeWhitespaces
</details>
Submitted a new commit. Changed the name and updated the call to reflect.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/platyps/116)*
<!-- Sent from Reviewable.io -->
